src/main/java/javaslang/collection/Stream.java

```diff
@@ -233,6 +233,41 @@ public T next() {
     }
 
     /**
+     * Returns a Stream containing {@code n} values of a given Function {@code f}
+     * over a range of integer values from 0 to {@code n - 1}.
+     *
+     * @param <T> Component type of the Stream
+     * @param n The number of elements in the Stream
+     * @param f The Function computing element values
+     * @return A Stream consisting of elements {@code f(0),f(1), ..., f(n - 1)}
+     * @throws NullPointerException if {@code n} or {@code f} are null
+     */
+    static <T> Stream<T> fill(Integer n, Function<Integer, ? extends T> f) {
+        Objects.requireNonNull(n, "n is null");
+        Objects.requireNonNull(f, "f is null");
+        int nOrZero = java.lang.Math.max(n, 0);
+        @SuppressWarnings("unchecked")
+        T[] elements = (T[]) new Object[nOrZero];
+        for (int i = 0; i < n; i++) {
+            elements[i] = f.apply(i);
+        }
+        return Stream.of(elements);
```

Comment:
Please make Stream.tabulate lazy: `return Stream.ofAll(Iterator.tabulate(n, f));`

You may add these methods to `javaslang.collection.Iterator`:

```java
    static <T> Iterator<T> fill(int n, Supplier<? extends T> s) {
        return tabulate(n, anything -> s.get());
    }

    static <T> Iterator<T> tabulate(int n, Function<? super Integer, ? extends T> f) {
        Objects.requireNonNull(f, "f is null");
        return new AbstractIterator<T>() {

            int i = 0;

            @Override
            public boolean hasNext() {
                return i < n;
            }

            @Override
            protected T getNext() {
                return f.apply(i++);
            }
        };
    }
```